src/k8s/api/v1/types.go

```diff
@@ -69,74 +88,29 @@ func (c ClusterStatus) HaClusterFormed() bool {
 
 // TICS -COV_GO_SUPPRESSED_ERROR
 // we are just formatting the output for the k8s status command, it is ok to ignore failures from result.WriteString()
-func (c ClusterStatus) datastoreToString() string {
-	result := strings.Builder{}
-
-	// Datastore
-	if c.Datastore.Type != "" {
-		result.WriteString(fmt.Sprintf("  type: %s\n", c.Datastore.Type))
-		// Datastore URL for external only
-		if c.Datastore.Type == "external" {
-			result.WriteString(fmt.Sprintln("  servers:"))
-			for _, serverURL := range c.Datastore.Servers {
-				result.WriteString(fmt.Sprintf("    - %s\n", serverURL))
-			}
-			return result.String()
-		}
-	}
-
-	// Datastore roles for dqlite
-	voters := make([]NodeStatus, 0, len(c.Members))
-	standBys := make([]NodeStatus, 0, len(c.Members))
-	spares := make([]NodeStatus, 0, len(c.Members))
-	for _, node := range c.Members {
-		switch node.DatastoreRole {
-		case DatastoreRoleVoter:
-			voters = append(voters, node)
-		case DatastoreRoleStandBy:
-			standBys = append(standBys, node)
-		case DatastoreRoleSpare:
-			spares = append(spares, node)
-		}
-	}
-	if len(voters) > 0 {
-		result.WriteString("  voter-nodes:\n")
-		for _, voter := range voters {
-			result.WriteString(fmt.Sprintf("    - %s\n", voter.Address))
-		}
-	} else {
-		result.WriteString("  voter-nodes: none\n")
-	}
-	if len(standBys) > 0 {
-		result.WriteString("  standby-nodes:\n")
-		for _, standBy := range standBys {
-			result.WriteString(fmt.Sprintf("    - %s\n", standBy.Address))
-		}
-	} else {
-		result.WriteString("  standby-nodes: none\n")
-	}
-	if len(spares) > 0 {
-		result.WriteString("  spare-nodes:\n")
-		for _, spare := range spares {
-			result.WriteString(fmt.Sprintf("    - %s\n", spare.Address))
-		}
-	} else {
-		result.WriteString("  spare-nodes: none\n")
-	}
-
-	return result.String()
-}
 
 // TODO: Print k8s version. However, multiple nodes can run different version, so we would need to query all nodes.
 func (c ClusterStatus) String() string {
 	result := strings.Builder{}
 
 	// Status
 	if c.Ready {
-		result.WriteString("status: ready")
+		result.WriteString("cluster status: ready")
 	} else {
-		result.WriteString("status: not ready")
+		result.WriteString("cluster status: not ready")
+	}
+	result.WriteString("\n")
+
+	// Control Plane Nodes
+	result.WriteString("control plane nodes: ")
+	addrMap := c.getCPNodeAddrToRoleMap()
+	nodes := make([]string, len(addrMap))
+	i := 0
+	for addr, role := range addrMap {
+		nodes[i] = fmt.Sprintf("%s (%s)", addr, role)
+		i++
 	}
+	result.WriteString(strings.Join(nodes, ", "))
```

Comment:
you can just loop over the `c.Members`, and `result.WriteString` them here? you have two extra loops that come with a lot of allocations each

Reply:
i did this because of the "," between IPs. 
this:
```go
for _, m := range c.Members {
    result.WriteString("%s (%s),", m.Address, m.DatastoreRole)
}
```
will result in something like:
`control plane nodes: 1.2.3.4 (voter), 2,3,4,5 (stand-by),` (notice the extra "," at the end)

we will have an extra separator or we can have none (separate them by spaces), correct?

Reply:
ok, how about something like this, then:

```go
members := make([]string, 0, len(c.Members))
for _, member := range c.Members {
  members = append(members, fmt.Sprintf("%s (%s)", member.Address, member.Role))
}

result.WriteString(strings.Join(members, ", ")
```

Reply:
or we can just separate them with 2 spaces and call it a day 
`192.168.0.1 (voter)  192.168.0.2 (voter)  192.168.0.3 (stand-by)`

Reply:
> or maybe in case the underlying array has enough space, append does not reallocate?

exactly, that's what `make([]T, 0, len)` does. `append` here will not do extra allocations